### PR TITLE
feat: settings network

### PR DESCRIPTION
**Pull request by fbwoolf**
This PR is branched off my theme settings branch. It wires up being able to change networks.

signal-2024-09-04-163006_003 signal-2024-09-04-163006_002

### CODE REVIEW

File: apps/mobile/src/app/wallet/settings/networks/index.tsx [hunk 1]
```diff
@@ -28,23 +38,26 @@ export default function SettingsNetworksScreen() {
         }}
       >
         <Cell
-          isSelected
+          isSelected={settings.network.id === 'mainnet'}
+          onPress={() => onPressChangeNetwork('mainnet')}
           title={t`Mainnet`}
-          subtitle={t`Enabled`}
+          subtitle={settings.network.id === 'mainnet' ? t`Enabled` : t`Disabled`}
```
Comment (pete-watters) [inline apps/mobile/src/app/wallet/settings/networks/index.tsx:44]: It seems like we could clean this up a bit but I think it's fine as is. 

Maybe destructure and renaming `settings.network.id` earlier? 

```
const { network: { id: networkId } } = settings
```

Or we could have a `switch` to pass props to `<Cell` based on `settings.network.id`

**Review by pete-watters — APPROVED** [review]
Nice work 👍 I added a suggestion but fine as is

File: apps/mobile/src/app/wallet/settings/networks/index.tsx (hunk 1 above)
Comment (kyranjamie) [reply apps/mobile/src/app/wallet/settings/networks/index.tsx:44]: Just a style preference, but isn't accessing a few properties deep of an object cleaner than a nested destructure?

On grounds that `{ network: { id: networkId } }` is more syntax heavy than `settings.network.id`?

**Review by kyranjamie — APPROVED** [review]

File: apps/mobile/src/app/wallet/settings/networks/index.tsx (hunk 1 above)
Comment (pete-watters) [reply apps/mobile/src/app/wallet/settings/networks/index.tsx:44]: I think it's cleaner to do one destructure to get what you need than multiple chains. Just my preference though

**Pull request merged**